course_catalog/etl/youtube_test.py

```diff
@@ -0,0 +1,189 @@
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube.yaml"
+
+    mocked_responses.add(
+        mocked_responses.GET,
+        settings.VIDEOS_CONFIG_FILE_LOCATION,
+        body=mock_channel_data,
+    )
+    yield mocked_responses
+
+
+@pytest.fixture
+def mocked_channel_empty_response(mocked_responses, settings):
+    """Mock the channel catalog response when the response is empty"""
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube_empty.yaml"
+
+    mocked_responses.add(
+        mocked_responses.GET, settings.VIDEOS_CONFIG_FILE_LOCATION, body=""
+    )
+    yield mocked_responses
+
+
+@pytest.mark.usefixtures("mocked_channel_response")
+def test_extract(mocker, mock_video_data, mock_video_batch_data, settings):
+    """Test youtube video ETL extract"""
+
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube.yaml"
+    settings.YT_DEVELOPER_KEY = "key"
+
+    mock_video_client = mocker.patch("course_catalog.etl.youtube.get_youtube_client")
+    mock_video_client.return_value.search.return_value.list.return_value.execute.side_effect = [
+        mock_video_search_data("token"),
+        mock_video_search_data(""),
+        mock_video_search_data(""),
+    ]
+
+    mock_video_client.return_value.videos.return_value.list.return_value.execute.return_value = (
+        mock_video_batch_data
+    )
+
+    assert list(youtube.extract()) == [
+        ("MIT", mock_video_data),
+        ("MIT", mock_video_data),
+        ("OCW", mock_video_data),
+    ]
+
+
+@pytest.mark.usefixtures("mocked_channel_response")
+def test_extract_with_bad_channel_id(
+    mocker, mock_video_data, mock_video_batch_data, settings
+):
+    """Test youtube video ETL extract with a channel for which the google api throws an error"""
+
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube.yaml"
+    settings.YT_DEVELOPER_KEY = "key"
+
+    mock_video_client = mocker.patch("course_catalog.etl.youtube.get_youtube_client")
+    mock_video_client.return_value.search.return_value.list.return_value.execute.side_effect = [
+        googleapiclient.errors.HttpError(MagicMock(), bytes()),
+        mock_video_search_data(""),
+    ]
+
+    mock_video_client.return_value.videos.return_value.list.return_value.execute.return_value = (
+        mock_video_batch_data
+    )
+
+    assert list(youtube.extract()) == [("OCW", mock_video_data)]
+
+
+def test_extract_with_unset_keys(settings):
+    """Test youtube video ETL extract with no keys set"""
+
+    settings.VIDEOS_CONFIG_FILE_LOCATION = None
+    settings.YT_DEVELOPER_KEY = None
+
+    assert list(youtube.extract()) == []
+
+
+@pytest.mark.usefixtures("mocked_channel_empty_response")
+def test_extract_with_no_channels(mocker, settings):
+    """Test youtube video ETL extract with no channels in data"""
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube_empty.yaml"
+    settings.YT_DEVELOPER_KEY = "key"
+    mocker.patch("course_catalog.etl.youtube.get_youtube_client")
```

Comment:
I think you actually want to patch `yaml.safe_load` instead and test that for `{}` and `None` return values. I think this test passes because the unmocked `yaml.safe_load` is returning `None` due to being passed a `Mock` which it can't parse.

This is a good use case for [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) which I don't think you've worked with yet.